File: 125-valid-palindrome/125-valid-palindrome.ts
function isPalindrome(s: string): boolean {
  s = s.replace(/[^a-zA-Z0-9]/g, '').toLowerCase();
  
  for (let i = 0, j = s.length - 1; i < j; i++, j--) {
    if (s[i] !== s[j]) {
      return false;
    }
  }
  
  return true;
};

File: 125-valid-palindrome/125-valid-palindrome.go
import "regexp"

func isPalindrome(s string) bool {
    re := regexp.MustCompile(`[^a-zA-Z0-9]`)
    s = re.ReplaceAllLiteralString(s, "")
    s = strings.ToLower(s)
    
    for i, j := 0, len(s) - 1; i < j; i, j = i + 1, j - 1 {
        if s[i] != s[j] {
            return false
        }
    }
    return true
}